### PR TITLE
Split transfer agreement page into two

### PR DESCRIPTION
**Pull request by techncl**

### CODE REVIEW

File: test/controllers/TransferAgreementController2Spec.scala [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+      headers(transferAgreementPage) mustBe TreeMap("Cache-Control" -> "no-store, must-revalidate")
+      transferAgreementPageAsString must include(
+        s"""href="/consignment/c2efd3e6-6664-4582-8c28-dcf891f60e68/upload">
+           |                Continue""".stripMargin)
+      transferAgreementPageAsString must include("You have already confirmed all statements")
+      taHelper.checkHtmlOfComplianceFormOptions.checkForOptionAndItsAttributes(transferAgreementPageAsString, formSuccessfullySubmitted = true)
+    }
+
+    "render the TA (compliance) 'already confirmed' page with an authenticated user if user navigates back to TA page" +
+      "after successfully submitting TA form that had been incorrectly submitted (empty) prior" in {
```
[the remark this answers is not in the document]
Comment (techncl) [reply test/controllers/TransferAgreementController2Spec.scala:211]: But this test is for when you've submitted a correct (fully filled in) form *after* previously submitting an incorrect (empty) one. What you've wrote sounds like the user has successfully submitted an empty form; unless I'm mistaken?

The reason for this test was that before when you submitted and incomplete form (which shows you the TA page with your mistakes), then submitted it correctly, when you went back to the page from the upload page, it wouldn't display the "Already confirmed" page, instead it would display the TA page with the errors you encountered on the first try.

File: test/controllers/TransferAgreementController2Spec.scala [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+      headers(taAlreadyConfirmedPage) mustBe TreeMap("Cache-Control" -> "no-store, must-revalidate")
+      taAlreadyConfirmedPageAsString must include(
+        s"""href="/consignment/c2efd3e6-6664-4582-8c28-dcf891f60e68/upload">
+           |                Continue""".stripMargin)
+      taAlreadyConfirmedPageAsString must include("You have already confirmed all statements")
+      taHelper.checkHtmlOfComplianceFormOptions.checkForOptionAndItsAttributes(taAlreadyConfirmedPageAsString, formSuccessfullySubmitted = true)
+    }
+
+    "render the TA (compliance) 'already confirmed' page with an authenticated user if user navigates back to TA page" +
+      "after successfully submitting TA form that had been incorrectly submitted (partially) prior" in {
```
[the remark this answers is not in the document]
Comment (techncl) [reply test/controllers/TransferAgreementController2Spec.scala:231]: Same comment as the one I left above

File: test/controllers/TransferAgreementController2Spec.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (TomJKing) [reply test/controllers/TransferAgreementController2Spec.scala]: Ah, OK, fair enough. Maybe reword it to: `... after successfully submiting TA form after previously sumbitting and empty form` What do you think? It's quite a hard description to write.

File: test/controllers/TransferAgreementController2Spec.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (TomJKing) [reply test/controllers/TransferAgreementController2Spec.scala]: OK. As my comment above, maybe reword slightly to: `... after successfully submitting TA form having previously submitted a partially complete form`. What do you think?

File: test/controllers/TransferAgreementController1Spec.scala [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+
+      playStatus(transferAgreementPage) mustBe OK
+      contentType(transferAgreementPage) mustBe Some("text/html")
+      headers(transferAgreementPage) mustBe TreeMap("Cache-Control" -> "no-store, must-revalidate")
+      transferAgreementPageAsString must include("Please confirm that the court judgment contains the following information.")
+      transferAgreementPageAsString must include(s"""<a href="/judgment/$consignmentId/upload"""" +
+           """ role="button" draggable="false" class="govuk-button" data-module="govuk-button">""")
+    }
+
+    "return forbidden if a judgment user submits a transfer agreement form" in {
```
[the remark this answers is not in the document]
Comment (techncl) [reply test/controllers/TransferAgreementController1Spec.scala]: Yes, I didn't add this but it seems the same to me too.

File: test/controllers/UploadControllerSpec.scala [hunk 4]
```diff
@@ -48,11 +48,11 @@ class UploadControllerSpec extends FrontEndTestHelper {
       val uploadPage = controller.uploadPage(consignmentId)
         .apply(FakeRequest(GET, "/consignment/1/upload").withCSRFToken)
       status(uploadPage) mustBe SEE_OTHER
-      redirectLocation(uploadPage).get must equal(s"/consignment/$consignmentId/transfer-agreement")
+      redirectLocation(uploadPage).get must equal(s"/consignment/$consignmentId/transfer-agreement1")
     }
 
     // This is unlikely but it's possible that they've bypassed the checks and partially agreed to things
```
Comment (TomJKing) [inline test/controllers/UploadControllerSpec.scala:54]: Can remove this comment

File: test/services/TransferAgreementServiceSpec.scala [hunk 5]
```diff
@@ -26,78 +28,131 @@ class TransferAgreementServiceSpec extends FlatSpec with Matchers with MockitoSu
...
+  when(graphQlConfig.getClient[atac.Data, atac.Variables]())
+    .thenReturn(graphQlClientForTACompliance) // Please ignore the Implicit-related error that IntelliJ displays, as it is incorrect.
 
   private val transferAgreementService: TransferAgreementService = new TransferAgreementService(graphQlConfig)
   private val consignmentId = UUID.fromString("e1ca3948-ee41-4e80-85e6-2123040c135d")
 
   private val token = new BearerAccessToken("some-token")
 
-  private val formData = TransferAgreementData(publicRecord = true,
+  private val taNotComplianceFormData = TransferAgreementData(publicRecord = true,
```
Comment (TomJKing) [inline test/services/TransferAgreementServiceSpec.scala:44]: Probably want to rename this, to match the naming scheme, ie with `beta`

File: test/services/TransferAgreementServiceSpec.scala [hunk 6]
```diff
@@ -26,78 +28,131 @@ class TransferAgreementServiceSpec extends FlatSpec with Matchers with MockitoSu
   implicit val ec: ExecutionContext = ExecutionContext.global
 
   private val graphQlConfig = mock[GraphQLConfiguration]
-  private val graphQlClient = mock[GraphQLClient[ata.Data, ata.Variables]]
-  when(graphQlConfig.getClient[ata.Data, ata.Variables]())
-    .thenReturn(graphQlClient) // Please ignore the Implicit-related error that IntelliJ displays, as it is incorrect.
+  private val graphQlClientForTANotCompliance = mock[GraphQLClient[atanc.Data, atanc.Variables]]
```
Comment (TomJKing) [inline test/services/TransferAgreementServiceSpec.scala:31]: Probably want to rename this to match the naming scheme, ie with `beta`

File: test/services/TransferAgreementServiceSpec.scala [hunk 7]
```diff
@@ -26,78 +28,131 @@ class TransferAgreementServiceSpec extends FlatSpec with Matchers with MockitoSu
...
     crownCopyright = true,
-    english = true,
+    english = true)
+
+  private val taComplianceFormData = TransferAgreementComplianceData(
     droAppraisalSelection = true,
     droSensitivity = true,
     openRecords = true)
+  
+  private val transferAgreementNotComplianceInput = AddTransferAgreementNotComplianceInput(consignmentId,
```
Comment (TomJKing) [inline test/services/TransferAgreementServiceSpec.scala:53]: Probably want to rename this, to match the naming scheme, ie with `beta`

File: test/util/TransferAgreementTestHelper.scala [hunk 8]
```diff
@@ -0,0 +1,179 @@
...
+import services.{ConsignmentService, TransferAgreementService}
+import uk.gov.nationalarchives.tdr.GraphQLClient
+import uk.gov.nationalarchives.tdr.GraphQLClient.Extensions
+
+import scala.concurrent.ExecutionContext
+
+class TransferAgreementTestHelper(wireMockServer: WireMockServer) extends FrontEndTestHelper{
+  implicit val ec: ExecutionContext = ExecutionContext.global
+
+  val nonComplianceOptions = Map(
```
Comment (TomJKing) [inline test/util/TransferAgreementTestHelper.scala:29]: Probably should rename this, in line with the naming scheme

File: app/services/TransferAgreementService.scala [hunk 9]
```diff
@@ -13,19 +14,33 @@ import scala.concurrent.{ExecutionContext, Future}
...
-  def addTransferAgreement(consignmentId: UUID, token: BearerAccessToken, formData: TransferAgreementData): Future[ata.AddTransferAgreement] = {
-    val addTransferAgreementInput: AddTransferAgreementInput = AddTransferAgreementInput(consignmentId,
+  private val addTransferAgreementNotComplianceClient =
+    graphqlConfiguration.getClient[atanc.Data, atanc.Variables]() // Please ignore the Implicit-related error that IntelliJ displays, as it is incorrect.
+  private val addTransferAgreementComplianceClient =
+    graphqlConfiguration.getClient[atac.Data, atac.Variables]() // Please ignore the Implicit-related error that IntelliJ displays, as it is incorrect.
+  def addTransferAgreementNotCompliance(consignmentId: UUID,
+                                        token: BearerAccessToken,
+                                        formData: TransferAgreementData): Future[atanc.AddTransferAgreementNotCompliance] = {
+    val addTransferAgreementNotComplianceInput: AddTransferAgreementNotComplianceInput = AddTransferAgreementNotComplianceInput(consignmentId,
```
Comment (TomJKing) [inline app/services/TransferAgreementService.scala:24]: See comment about, about renaming.

**Review by TomJKing — COMMENTED** [review]
Looks good. Sorry, there are just comments about keeping the naming conventions consistent across the code and tests, ie ensuring that the `beta` naming convention is used, instead of 'non-compliance`. I think I've commented on most of them, but check through the code to make sure its all consistent. 

Thanks for persisting with this, I know its been a bit of a pain to implement. Almost there.

File: app/services/TransferAgreementService.scala [hunk 10]
```diff
@@ -13,19 +14,33 @@ import scala.concurrent.{ExecutionContext, Future}
...
 class TransferAgreementService @Inject()(val graphqlConfiguration: GraphQLConfiguration) (implicit val ec: ExecutionContext){
-  private val addTransferAgreementClient =
-    graphqlConfiguration.getClient[ata.Data, ata.Variables]() // Please ignore the Implicit-related error that IntelliJ displays, as it is incorrect.
-  def addTransferAgreement(consignmentId: UUID, token: BearerAccessToken, formData: TransferAgreementData): Future[ata.AddTransferAgreement] = {
-    val addTransferAgreementInput: AddTransferAgreementInput = AddTransferAgreementInput(consignmentId,
+  private val addTransferAgreementNotComplianceClient =
+    graphqlConfiguration.getClient[atanc.Data, atanc.Variables]() // Please ignore the Implicit-related error that IntelliJ displays, as it is incorrect.
+  private val addTransferAgreementComplianceClient =
+    graphqlConfiguration.getClient[atac.Data, atac.Variables]() // Please ignore the Implicit-related error that IntelliJ displays, as it is incorrect.
+  def addTransferAgreementNotCompliance(consignmentId: UUID,
```
[the remark this answers is not in the document]
Comment (techncl) [reply app/services/TransferAgreementService.scala:21]: Ah 🤦 . I forgot about these names.

File: test/controllers/TransferAgreementComplianceControllerSpec.scala [hunk 11]
```diff
@@ -0,0 +1,283 @@
...
+  override def afterEach(): Unit = {
+    wiremockServer.resetAll()
+    wiremockServer.stop()
+  }
+
+  val taHelper = new TransferAgreementTestHelper(wiremockServer)
+
+  "TransferAgreementComplianceController GET" should {
+
+    "redirect to the transfer agreement not-compliance page with an authenticated user if consignment status is 'None'" in {
```
[the remark this answers is not in the document]
Comment (techncl) [reply test/controllers/TransferAgreementComplianceControllerSpec.scala:38]: So 
1. "transfer agreement (transfer agreement)" & "transfer agreement (transfer agreement continued)"
or
2. "transfer agreement" & "transfer agreement  (continued)"?

File: test/controllers/TransferAgreementComplianceControllerSpec.scala [hunk 12]
```diff
@@ -0,0 +1,283 @@
...
+      val transferAgreementPageAsString = contentAsString(transferAgreementSubmit)
+
+      playStatus(transferAgreementSubmit) mustBe BAD_REQUEST
+      taHelper.checkHtmlOfComplianceFormOptions.checkForOptionAndItsAttributes(transferAgreementPageAsString, incompleteTransferAgreementForm.toMap)
+      transferAgreementPageAsString must include("govuk-error-message")
+      transferAgreementPageAsString must include("error")
+      taHelper.checkHtmlContentForErrorSummary(transferAgreementPageAsString, taHelper.compliance, Set())
+    }
+
+    "display errors when a partially complete compliance form is submitted" in {
```
[the remark this answers is not in the document]
Comment (techncl) [reply test/controllers/TransferAgreementComplianceControllerSpec.scala:176]: Same as above. here, only 2 options are being selected, as seen on line 181.

File: test/controllers/TransferAgreementPrivateBetaControllerSpec.scala [hunk 13]
```diff
@@ -0,0 +1,316 @@
...
+      val transferAgreementPageAsString = contentAsString(transferAgreementSubmit)
+
+      playStatus(transferAgreementSubmit) mustBe BAD_REQUEST
+      taHelper.checkHtmlOfNonComplianceFormOptions.checkForOptionAndItsAttributes(transferAgreementPageAsString, incompleteTransferAgreementForm.toMap)
+      transferAgreementPageAsString must include("govuk-error-message")
+      transferAgreementPageAsString must include("error")
+      taHelper.checkHtmlContentForErrorSummary(transferAgreementPageAsString, taHelper.notCompliance, Set())
+    }
+
+    "display errors when a partially complete not-compliance form is submitted" in {
```
[the remark this answers is not in the document]
Comment (techncl) [reply test/controllers/TransferAgreementPrivateBetaControllerSpec.scala:159]: Please see [this comment](https://github.com/nationalarchives/tdr-transfer-frontend/pull/1343#discussion_r803978643) above

File: test/controllers/TransferAgreementComplianceControllerSpec.scala (hunk 12 above)
[the remark this answers is not in the document]
Comment (TomJKing) [reply test/controllers/TransferAgreementComplianceControllerSpec.scala:176]: Fair enough. My mistake!

**Review by TomJKing — APPROVED** [review]
Looks good. Thanks for wading through this. It's been a bit of a mission to do this work!
